### PR TITLE
Check recipe flags at schedule creation (including fix for youzim.it)

### LINKED ISSUE #864: Schedule configuration flags must also be validated at schedule creation
- Location: API

## Problem

Schedule configuration flags have various constraints which must be checked. These constraints are check when the schedule configuration is updated (`PATCH /schedule/{schedule_id}`) but not when the schedule is created from scratch (`POST /schedule`). 

This has almost zero impact on farm.openzim.org since the UI does not allows to create schedules from scratch but it has a very significant one on youzim.it where the UI is always creating schedules from scratch (and most probably, same issue impact WP1, but I'm less familiar with it).

Once fixed, this will solve (first part of) https://github.com/openzim/zimit-frontend/issues/40

## Reproducing steps

Create (not clone and patch) a new schedule with a very long description (longer than 80 chars). It will succeed while it should fails.

### LINKED ISSUE #867: zimit offliner: allow to disable the ZIM filename check
In youzim.it, the ZIM filename does not obey our naming convention. 

As discussed in https://github.com/openzim/zimit-frontend/issues/43, it must hence be possible to disable the check with an environment variable (to still activate the check on farm.openzim.org).

### PR DESCRIPTION
**Pull request by benoit74**
## Rationale

Fix #864 (again)
Fix #867 (mandatory to deploy #864 on youzim.it)

## Changes
- new environment variable ~`ZIMIT_DISABLE_ZIM_FILENAME_CHECK`~ `ZIMIT_USE_RELAXED_SCHEMA` which must be set to disable the ZIM filename check
- new logic to use this variable to select a relaxed offliner schema definition
- previous changes from https://github.com/openzim/zimfarm/pull/865

### CODE REVIEW

**Comment by benoit74** [comment]
Nota: do not merge until decision has been taken on https://github.com/openzim/zimit-frontend/issues/42 + synchronization with Travis is needed

File: dispatcher/backend/src/common/constants.py
Comment (rgaudin) [inline dispatcher/backend/src/common/constants.py:102]: Should be a `bool`, we don't want to carry whatever was passed to this env

File: dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py
Comment (rgaudin) [reply dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py:109]: Would it be possible (while remaining clear and easy to maintain) to alter schema in `get_offliner_schema` so that the flag becomes _dynamic_?

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thank you ; please take a look at suggestions and question

File: dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py
Comment (benoit74) [reply dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py:109]: > Change your test to have different expectations based on the value of the flag instead of keeping a commented copy that will never be used.

Did I get it correctly that you mean that we will still have to modify flag value more or less manually to check both test cases, but at least we won't have commented code anymore? This makes sense to me, but I'm not sure I got you correctly.

Comment (benoit74) [reply dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py:109]: Never mind, I achieved to make both test run ok, my bad Python knowledge was the cultprit. Same solution will probably apply to the other PR about worker IPs, do not review it for now, I will fix it and keep you informed.

Comment (rgaudin) [reply dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py:109]: OK

**Comment by benoit74** [comment]
Following your review, I made following changes:
- create a `ZimitFlagsSchemaRelaxed` class which override the `zim_file` field to remove the validation
- use conditionally the `ZimitFlagsSchemaRelaxed` in `get_offliner_schema`
- renamed the environment variable to better match its "new" purpose + transformed into a bool
- fixed tests so that all cases are now covered by manipulating the constant (manipulating the constant is where my bad Python programming made me fail previously)

File: dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py
Comment (rgaudin) [inline dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py:26]: A namedtuple would serve the same purpose and be less verbose

File: dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py
Comment (rgaudin) [inline dispatcher/backend/src/tests/integration/routes/schedules/test_zimit.py:9]: A dosctring would help

**Review by rgaudin — APPROVED** [review]
LGTM ; couple minor suggestions. I'll let you decide what to do. Please merge yourself

**Comment by benoit74** [comment]
Suggested changes made, makes a lot of sense.

I converted this PR to Draft so that it is not merged by accident, we need to synchronize with Travis + prepare new env variable for youzim.it

**Pull request merged**
